server/events/vcs/gitlab_client.go

```diff
@@ -49,6 +49,11 @@ func NewGitlabClient(hostname string, token string, logger *logging.SimpleLogger
 
 	// If not using gitlab.com we need to set the URL to the API.
 	if hostname != "gitlab.com" {
+		//URL.Parse only works on URLs (i.e.: we need a scheme)
+		if !strings.HasPrefix(hostname, "http://") && !strings.HasPrefix(hostname, "https://") {
+			hostname = "https://" + hostname
+		}
+
 		u, err := url.Parse(hostname)
 		if err != nil {
 			return nil, errors.Wrapf(err, "parsing hostname %q", hostname)
```

Comment:
and here

Reply:
The code below here is redundant now:
```go

 		scheme := "https"
 		if u.Scheme != "" {
 			scheme = u.Scheme
 		}
 		apiURL := fmt.Sprintf("%s://%s/api/v4/", scheme, u.Host)
```

It should instead read:
```go
	// Now we're ready to construct the client.
	hostname = strings.TrimSuffix(hostname, "/")
	apiURL := fmt.Sprintf("%s/api/v4/", hostname)
	if err := client.Client.SetBaseURL(apiURL); err != nil {
		return nil, errors.Wrapf(err, "setting GitLab API URL: %s", apiURL)
	}
```
Which additionally allows gitlab to be hosted at a base path.